worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/adapter/Regenerator.java

```diff
@@ -0,0 +1,559 @@
+                for (int i = 0; minz + i * numlists + offset <= maxz; i++)
+                    para.add(byz.get(minz + i * numlists + offset));
+                tasks.add(para);
+            }
+        }
+
+        return tasks;
+    }
+
+    protected static Random getChunkRandom(long worldseed, int x, int z) {
+        Random random = new Random();
+        random.setSeed(worldseed);
+        long xRand = random.nextLong() / 2L * 2L + 1L;
+        long zRand = random.nextLong() / 2L * 2L + 1L;
+        random.setSeed((long) x * xRand + (long) z * zRand ^ worldseed);
+        return random;
+    }
+
+    //util
+    public static void time(Runnable r, String text) {
+        long starttask = System.currentTimeMillis();
+//            System.out.println(text);
+        r.run();
+        //Debug outputs:
+        //System.out.println(text + " took " + (System.currentTimeMillis() - starttask) + "ms");
+    }
+
+    //classes
+    public static abstract class ChunkStatusWrapper<IChunkAccess> {
+
+        /**
+         * Must not return a negative integer
+         *
+         * @return the radius of required neighbor chunk
+         */
+        public abstract int requiredNeigborChunkRadius();
+
+        public abstract String name();
+
+        public abstract void processChunk(Long xz, List<IChunkAccess> accessibleChunks);
+
+        void processChunkSave(Long xz, List<IChunkAccess> accessibleChunks) {
+            try {
+                processChunk(xz, accessibleChunks);
+            } catch (Exception e) {
+                System.err.println("Error while running " + name() + " on chunk " + MathMan.unpairIntX(xz) + "/" + MathMan.unpairIntY(xz));
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public enum Concurrency {
+        FULL,
+        RADIUS,
+        NONE
+    }
+
+    public static class SequentialTasks<T> extends Tasks<T> {
+
+        public SequentialTasks(int expectedsize) {
+            super(expectedsize);
+        }
+    }
+
+    public static class ConcurrentTasks<T> extends Tasks<T> {
+
+        public ConcurrentTasks(int expectedsize) {
+            super(expectedsize);
+        }
+    }
+
+    public static class Tasks<T> implements Iterable<T> {
+
+        private final List<T> tasks;
+
+        public Tasks(int expectedsize) {
+            tasks = new ArrayList(expectedsize);
+        }
+
+        public void add(T task) {
+            tasks.add(task);
+        }
+
+        public List<T> list() {
+            return tasks;
+        }
+
+        public int size() {
+            return tasks.size();
+        }
+        
+        @Override
+        public Iterator<T> iterator() {
+            return tasks.iterator();
+        }
+
+        @Override
+        public String toString() {
+            return toString(0);
+        }
+        
+        public String toString(int indent) {
+            String ret = new String(new char[indent]).replace('\0', ' ') + getClass().getSimpleName() +" [\n";
```

Comment:
Could use `Arrays.fill` instead of that replace

Reply:
done